src/addagent/validate.c

```diff
@@ -515,19 +523,45 @@ double OS_AgentAntiquity_ID(const char *id) {
     return ret;
 }
 
-/* Returns the number of seconds since last agent connection, or -1 if error. */
+/**
+ * @brief Returns the number of seconds since last agent connection
+ * 
+ * @param name The name of the agent
+ * @param ip The IP address of the agent (unused). Kept only for compatibility
+ * @retval On success, it returns the difference between the current time and the last keepalive
+ * @retval -1 On error: invalid DB query syntax or result
+ */
 double OS_AgentAntiquity(const char *name, const char *ip)
 {
-    struct stat file_stat;
-    char file_name[OS_FLSIZE];
+    char wdbquery[OS_MAXSTR] = "";
+    char wdboutput[OS_MAXSTR] = "";
+    int wdb_sock = -1;
+    time_t output = 0;
+    cJSON *root = NULL;
+    cJSON *keepalive = NULL;
 
-    snprintf(file_name, OS_FLSIZE - 1, "%s/%s-%s", AGENTINFO_DIR, name, ip);
+    if(!name || !ip){
+        mdebug1("Empty agent name or ip when trying to get last keepalive. Agent: (%s) IP: (%s)", name, ip);
+        return OS_INVALID;
+    }
 
-    if (stat(file_name, &file_stat) < 0)
-        return -1;
+    snprintf(wdbquery, sizeof(wdbquery), SQL_SELECT_KEEPALIVE, name, ip, ip);
+    root = wdbc_query_parse_json(&wdb_sock, wdbquery, wdboutput, sizeof(wdboutput));
+
+    if (!root) {
+        merror("Error querying Wazuh DB to get the last agent keepalive.");
+        return OS_INVALID;
+    }
 
-    return difftime(time(NULL), file_stat.st_mtime);
+    keepalive = cJSON_GetObjectItem(cJSON_GetArrayItem(root, 0),"last_keepalive");
+    output = !keepalive ? 0 : keepalive->valueint;
+
+    cJSON_Delete(root);
+    os_free(keepalive);
+
+    return difftime(time(NULL), output);
 }
+#endif
```

Comment:
Code style: Add /* !CLIENT */ comment on closing large (multiple functions) **#def** blocks

Reply:
True, but no longer necessary after moving the method to **wdb_agent.c**

Reply:
Done!